### ISSUE TITLE
Make ttl and sortindex handling sane

### ISSUE THREAD

**Issue by data-sync-user** [mozilla/uniffi-rs#766]
ttl and sortindex are handled in a non-obvious and hacky way. History sync does this, and we are copying that for the clients record. We should fix it - but it's not obvious how - it probably needs a bit of a rethink about a payload vs a bso_record etc.

https://github.com/mozilla/application-services/blob/04f2e0ee8ba94fc27509078381102a7b0f793ab6/components/places/src/history_sync/record.rs#L32

**Comment by data-sync-user** [comment]
➤ Thom Chiovoloni commented:

Honestly, I can think of a few ways of doing this but they all seem like more trouble than they're worth. It doesn't really seem like that big of a footgun to me, but I guess it is a bit hacky.

**Comment by data-sync-user** [comment]
➤ Mark Hammond commented:

It just shot me in the foot 😀 The patch for only uploading the client record if it changed relies on the derived Eq. The patch started failing when I added ttl, because the ttl was different between the 2 records, even though they should be considered equal. I had to print the records to figure this out. I think the fact it shot me within 24 hours is a reasonable anecdote in support of it being a foot-gun 😀
